lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1378,21 +1379,21 @@ selectTxHistory ti wid minWithdrawal order conditions = do
     selectLatestCheckpoint wid >>= \case
         Nothing -> pure []
         Just cp -> do
-            minWithdrawalFilter <- case minWithdrawal of
-                Nothing  -> pure []
+            minWithdrawalPred <- case minWithdrawal of
+                Nothing -> pure (const True)
                 Just inf -> do
+                    -- Tests whether a given TxMeta matches a reward withdrawal
+                    -- above the given minimum. This filtering is done outside
+                    -- of sqlite because joins are not possible with persistent.
                     let coin = W.Coin $ fromIntegral $ getQuantity inf
-                    wdrls <- fmap entityVal
-                        <$> selectList [ TxWithdrawalAmount >=. coin ] []
-                    pure [ TxMetaTxId <-. (txWithdrawalTxId <$> wdrls) ]
-
-            metas <- fmap entityVal <$> selectList
-                ( mconcat
-                    [ [ TxMetaWalletId ==. wid ]
-                    , minWithdrawalFilter
-                    , conditions
-                    ]
-                ) sortOpt
+                    wdrls <- fmap entityVal <$> selectList
+                        [ TxWithdrawalAmount >=. coin ]
+                        [ Asc TxWithdrawalTxId ]
+                    let txids = Set.fromAscList $ map txWithdrawalTxId wdrls
+                    pure ((`Set.member` txids) . txMetaTxId)
+
+            metas <- filter minWithdrawalPred . fmap entityVal
+                <$> selectList ((TxMetaWalletId ==. wid):conditions) sortOpt
```

Comment:
Erf. This means that requests like `/transactions?min_withdrawal=1` will have to load the entire transaction history in memory :/  
The case reached by the state-machine is probably a case where there are more than 1000 withdrawals in the wallet right :thinking: ? So we could re-use a similar approach to what was done for fetching inputs / outputs:

```hs
        resolvedInputs <- toOutputMap . fmap entityVal <$>
            combineChunked inputs (\inputsChunk -> selectList
                [TxOutputTxId <-. (txInputSourceTxId <$> inputsChunk)]
                [Asc TxOutputTxId, Asc TxOutputIndex])
```

where `inputs` here would be `txids`. So we can fetch all transactions from the list of `txids` that match the extra conditions, and do it chunk by chunk.

Reply:
Yes - not ideal. If using `combineChunked` we also need to sort again after concatenating chunks. But that would be less bad.

Reply:
I expect most wallets to have a fairly limited amount of withdrawals. And, in case one has many, fetching all transactions first would be way worse.